### PR TITLE
Login tips

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

Added to conneting window tips.

## Why / Balance

Because I want

## Technical details

Well, like, I copypasted the tips code into the connection window code...

## Media

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: VigersRay
- add: Added loggin tips.

### CODE REVIEW

File: Content.Client/Launcher/LauncherConnectingGui.xaml.cs [hunk 1]
```diff
@@ -51,6 +67,12 @@ public LauncherConnectingGui(LauncherConnecting state)
             var edim = IoCManager.Resolve<ExtendedDisconnectInformationManager>();
             edim.LastNetDisconnectedArgsChanged += LastNetDisconnectedArgsChanged;
             LastNetDisconnectedArgsChanged(edim.LastNetDisconnectedArgs);
+            _cfg.OnValueChanged(CCVars.LoginTipsDataset, SetLoginTipsDataset, true);
+        }
```
Comment (metalgearsloth) [inline Content.Client/Launcher/LauncherConnectingGui.xaml.cs:71]: Events need to be unsubscribed or they leak, especially in a UI control.

Comment (VigersRay) [reply Content.Client/Launcher/LauncherConnectingGui.xaml.cs]: okay i use UnsubValueChanged

File: Content.Client/Launcher/LauncherConnectingGui.xaml.cs [hunk 2]
```diff
@@ -15,18 +21,28 @@ public sealed partial class LauncherConnectingGui : Control
     {
         private const float RedialWaitTimeSeconds = 15f;
         private readonly LauncherConnecting _state;
+        private readonly IRobustRandom _random;
+        private readonly IPrototypeManager _prototype;
+        private readonly IConfigurationManager _cfg;
+
         private float _redialWaitTime = RedialWaitTimeSeconds;
+        private string _loginTipsDataset = "Tips";
```
Comment (mirrorcult) [inline Content.Client/Launcher/LauncherConnectingGui.xaml.cs:29]:
```suggestion
        [ValidatePrototypeId<DatasetPrototype>]
        private string _loginTipsDataset = "Tips";
```

**Review by mirrorcult — DISMISSED** [review]

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
- Dummy control is unnecessary to parent panelcontainer too.
- You have both the dataset as a validateprototype and as a cvar.
- Login not Logit.

File: Content.Client/Launcher/LauncherConnectingGui.xaml.cs [hunk 3]
```diff
@@ -67,6 +90,30 @@ private void LastNetDisconnectedArgsChanged(NetDisconnectedArgs? args)
...
+            LogitTips.Visible = loginTipsEnable;
+            if (!loginTipsEnable)
+            {
+                return;
+            }
+
+            if (!_prototype.TryIndex<DatasetPrototype>(_loginTipsDataset, out var tips))
+                return;
+
+            var tipList = tips.Values.ToList();
```
Comment (metalgearsloth) [inline Content.Client/Launcher/LauncherConnectingGui.xaml.cs:105]: This ToList is unnecessary, it's already a list.

**Review by metalgearsloth — APPROVED** [review]

**Comment by VigersRay** [comment]
happy-clap

**Pull request merged**